### PR TITLE
Show submitted referrals

### PR DESCRIPTION
**Pull request by libuk**
# Context
In a previous change[1], we hid submitted referrals due to it causing
confusion for users. We were linking the user back to the check your
answers page which gave the impression that the application wasn't
submitted.

We are now going to link these referrals to a new page which shows the
full submitted referral.

[1]
https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/pull/516

# Changes in this PR
* Reinstate the submitted referrals tab
* Introduce full submitted referral page

### Before
Screenshot 2023-10-05 at 12 10 55

### After
Screenshot 2023-10-05 at 12 47 48
Screenshot 2023-10-05 at 12 03 16
Screenshot 2023-10-05 at 12 03 30

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by libuk** [comment]
[A referrer can view their own application that they've submitted](https://trello.com/c/CQgcsom7/1348-a-referrer-can-view-their-own-application-that-theyve-submitted)

**Comment by libuk** [comment]
[CPPs want to see their submitted referrals](https://trello.com/c/RTTZyBA4/1586-cpps-want-to-see-their-submitted-referrals)

File: server/controllers/apply/applicationsController.ts [hunk 1]
```diff
@@ -135,4 +135,17 @@ export default class ApplicationsController {
       return res.render('applications/confirm')
     }
   }
+
+  full(): RequestHandler {
```
Comment (pezholio) [inline server/controllers/apply/applicationsController.ts:139]: I tend to prefer resource-style routes (`index`, `show`, `update`, `create` etc), but I understand it's a pain sometimes to have to add a whole new controller for one action!

**Review by pezholio — APPROVED** [review]
LGTM 👍

File: server/controllers/apply/applicationsController.ts (hunk 1 above)
Comment (libuk) [reply server/controllers/apply/applicationsController.ts:139]: Ah yeah, it's a good point. As we have a similar route for [assessments](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/blob/main/server/controllers/temporary-accommodation/manage/assessmentsController.ts#L121) for an identical page, it's probably best we keep them in line for now. But I'll keep this in mind going forward 👍🏾

**Comment by selina-dxw** [comment]
Looks good, thanks!

**Pull request merged**